src/resultify.ts

```diff
@@ -0,0 +1,10 @@
+import { Result } from './result';
+
+export { resultify };
+
+type Callback<TArgs extends unknown[], TResult> = (...args: TArgs) => TResult;
```

Comment:
from testing this fn in 2 bigger codebases - the return typ need to be:
`TResult | Promise<TResult>` to fix some rare corner cases.